src/main/java/org/wildfly/security/auth/server/RealmIdentity.java

```diff
@@ -266,4 +284,28 @@ public boolean exists() throws RealmUnavailableException {
             return false;
         }
     };
+
+    /**
+     * Represents a key that is specific to the realm and used to uniquely identify a {@link RealmIdentity} instance.
+     */
+    interface Key {
+
+        Key EMPTY = (Key) () -> null;
+
+        /**
+         * Returns the value of this key.
+         *
+         * @return the value of this key
+         */
+        Object getValue();
```

Comment:
In part yes, and ideally also to enforce it via abstract methods if possible.  As it is right now - ironically it's essentially the same as Principal.